Slim/Web/Pages/Search.pm

```diff
@@ -592,7 +594,10 @@ sub fillInSearchResults {
 	# This is very similar to a loop in Slim::Web::Pages::BrowseDB....
 	while (my $obj = $rs->next) {
 
-		$obj->store_column('work' => $obj->work->get_column('title')) if $type eq 'track' && $obj->work->get_column('title');
+		# ensure work name is in the place expected by TitleFormatter so that it uses name and not id
+		if ( $type eq 'track' && $obj->workid && $obj->work && $obj->work->title ) {
+			$obj->store_column('work' => $obj->work->title);
```

Comment:
Hold on... aren't you changing the data object here?!? Hereafter the `work` wouldn't be the work object any more but only its name, right? That sounds wrong to me. Shouldn't the TitleFormatter (or whatever is consuming that data) be fixed to access the work's title instead?

Reply:
No, it's changing the object's representation of  tracks.work, that's all. See debug output below. I'm using `store_column` rather than `set_column` so DBIx doesn't think there's a pending change to write to the database. 

However, having just had a closer look at Slim::Music::Titleformatter, there is a mechanism for providing specific code for a format option, eg album name, which is the same problem I attempted to solve with store_column, so I'll go that way.

Before:
```
...
                            urlmd5             => "7c177e2cb8b846148547d02acfcee742",
                            vbr_scale          => 1,
                            virtual            => undef,
                            work               => 43_796,
                            year               => 1993,
                          },
    _in_storage        => 1,
    _inflated_column   => {
                            work => bless({
                                  _column_data   => {
                                                      composer => 528_211,
                                                      id => 43_796,
                                                      title => "Swan Lake - Suite, Op. 20",
                                                      titlesearch => "SWAN LAKE SUITE OP 20",
                                                      titlesort => "SWAN LAKE SUITE OP 0020",
                                                    },
                                  _in_storage    => 1,
                                  _source_handle => bless({ schema => "Slim::Schema", source_moniker => "Work" }, "DBIx::Class::ResultSourceHandle"),
                                }, "Slim::Schema::Work"),
                          },
...

```
After:
```

...
                            urlmd5             => "7c177e2cb8b846148547d02acfcee742",
                            vbr_scale          => 1,
                            virtual            => undef,
                            work               => "Swan Lake - Suite, Op. 20",
                            year               => 1993,
                          },
    _in_storage        => 1,
    _inflated_column   => {
                            work => bless({
                                  _column_data   => {
                                                      composer => 528_211,
                                                      id => 43_796,
                                                      title => "Swan Lake - Suite, Op. 20",
                                                      titlesearch => "SWAN LAKE SUITE OP 20",
                                                      titlesort => "SWAN LAKE SUITE OP 0020",
                                                    },
                                  _in_storage    => 1,
                                  _source_handle => bless({ schema => "Slim::Schema", source_moniker => "Work" }, "DBIx::Class::ResultSourceHandle"),
                                }, "Slim::Schema::Work"),
                          },
...
```

Reply:
Change pushed. It's worth noting that this results in an extra database call for each row in the tracks result set.